packages/main/src/Token.ts

```diff
@@ -0,0 +1,243 @@
+import slot from "@ui5/webcomponents-base/dist/decorators/slot.js";
+import event from "@ui5/webcomponents-base/dist/decorators/event.js";
+import customElement from "@ui5/webcomponents-base/dist/decorators/customElement.js";
+import languageAware from "@ui5/webcomponents-base/dist/decorators/languageAware.js";
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+import { getTheme } from "@ui5/webcomponents-base/dist/config/Theme.js";
+import {
+	isBackSpace,
+	isSpace,
+	isDelete,
+	isSpaceCtrl,
+} from "@ui5/webcomponents-base/dist/Keys.js";
+import "@ui5/webcomponents-icons/dist/decline.js";
+import "@ui5/webcomponents-icons/dist/sys-cancel.js";
+import { ITabbable } from "@ui5/webcomponents-base/dist/delegate/ItemNavigation.js";
+import type I18nBundle from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import { getI18nBundle } from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import { TOKEN_ARIA_DELETABLE } from "./generated/i18n/i18n-defaults.js";
+
+import Icon from "./Icon.js";
+import TokenTemplate from "./generated/templates/TokenTemplate.lit.js";
+
+// Styles
+import styles from "./generated/themes/Token.css.js";
+
+/**
+ * @class
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ * Tokens are small items of information (similar to tags) that mainly serve to visualize previously selected items.
+ *
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import "@ui5/webcomponents/dist/Token.js";</code>
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webc.main.Token
+ * @extends sap.ui.webc.base.UI5Element
+ * @tagname ui5-token
+ * @since 1.0.0-rc.9
+ * @implements sap.ui.webc.main.IToken
+ * @public
+ */
+@customElement("ui5-token")
+@languageAware
+
+/**
+ * Fired when the the component is selected by user interaction with mouse or by clicking space.
+ *
+ * @event sap.ui.webc.main.Token#select
+ * @public
+ */
+@event("select")
+
+/**
+ * Fired when the backspace, delete or close icon of the token is pressed
+ *
+ * @event
+ * @param {Boolean} backSpace Indicates whether token is deleted by backspace key.
+ * @param {Boolean} delete Indicates whether token is deleted by delete key.
+ * @private
+ */
+@event("delete", {
+	detail: {
+		"backSpace": { type: Boolean },
+		"delete": { type: Boolean },
+	},
+})
+
+class Token extends UI5Element implements ITabbable {
+	/**
+	 * Defines the text of the token.
+	 *
+	 * @type {string}
+	 * @name sap.ui.webc.main.Token.prototype.text
+	 * @defaultvalue ""
+	 * @public
+	 */
+	@property()
+	text!: string;
+
+	/**
+	 * Defines whether the component is read-only.
+	 * <br><br>
+	 * <b>Note:</b> A read-only component can not be deleted or selected,
+	 * but still provides visual feedback upon user interaction.
+	 *
+	 * @type {boolean}
+	 * @name sap.ui.webc.main.Token.prototype.readonly
+	 * @defaultvalue false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	readonly!: boolean;
+
+	/**
```

Comment:
Could you reorder the props and move the private ones after the public ones

Reply:
Done